File: 83/document-collab-frontend/src/api/document.ts
import { get, post, put, del } from './request'

export interface DocumentCreateDTO {
  name: string
  description?: string
}

export interface DocumentDTO {
  id: string
  tenantId: string
  name: string
  description?: string
  currentVersionId?: string
  currentVersionNumber?: number
  version: number
  createdBy: string
  createdAt: string
  updatedAt: string
}

export const documentApi = {
  create: (data: DocumentCreateDTO) => post<DocumentDTO>('/documents', data),
  getList: () => get<DocumentDTO[]>('/documents'),
  getById: (id: string) => get<DocumentDTO>(`/documents/${id}`),
  update: (id: string, name: string, description?: string) =>
    put<DocumentDTO>(`/documents/${id}?name=${encodeURIComponent(name)}${description ? '&description=' + encodeURIComponent(description) : ''}`),
  delete: (id: string) => del(`/documents/${id}`),
}

File: 83/document-collab-backend/src/main/java/com/doccollab/entity/Document.java
package com.doccollab.entity;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.CompoundIndexes;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "documents")
@CompoundIndexes({
        @CompoundIndex(name = "tenant_name_unique", def = "{'tenantId': 1, 'name': 1}", unique = true)
})
public class Document {

    @Id
    private String id;

    private String tenantId;

    private String name;

    private String description;

    private String currentVersionId;

    private Long version;

    private String createdBy;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}
